### ISSUE TITLE
Abstract out functionality from primarydock.cpp into include files.

### ISSUE THREAD

**Issue by ssepeq** [primaryodors/primarydock#196]
It's pretty spaghettified in there and should be better modularized.

**Comment by primaryodors** [comment]
Certain obsolete functions will be removed in the near future. It is best to postpone this issue until at least the following features are removed:

- Soft dock;
- Active helix modifications.

Ideally, the work for #313 should be completed before dividing up the code in `src/primarydock.cpp`.
